cms_account_form/models/account_form.py

```diff
@@ -0,0 +1,157 @@
+        'country_id',
+        'phone',
+        'email',
+        'website',
+    )
+    _form_fields_order = _form_model_fields
+    _form_required_fields = (
+        "name", "street", "zip", "city",
+        "country_id", "phone", "email"
+    )
+
+    @property
+    def form_title(self):
+        return _('My account')
+
+    @property
+    def form_msg_success_updated(self):
+        return _('Profile updated.')
+
+    def form_next_url(self, main_object=None):
+        if self.request.args.get('redirect'):
+            # redirect overridden
+            return self.request.args.get('redirect')
+        return '/my/home'
+
+    def form_update_fields_attributes(self, _fields):
+        """Set warning message for email updates."""
+        super().form_update_fields_attributes(_fields)
+        tmpl = self.env.ref('cms_account_form.email_field_update_warning')
+        _fields['email']['help'] = tmpl.render({})
+
+    def form_validate_email(self, value, **req_values):
+        error, message = None, None
+        if value and not validate_email(value):
+            error = 'email_not_valid'
+            message = _('Invalid Email! Please enter a valid email address.')
+        return error, message
+
+    def form_validate_vat(self, value, **req_values):
+        error, message = None, None
+        if value and hasattr(self.form_model, 'check_vat'):
+            country_id = int(req_values.get("country_id", 0))
+            if country_id:
+                value = self.form_model.fix_eu_vat_number(country_id, value)
+            partner_dummy = self.form_model.new({
+                'vat': value,
+                'country_id': country_id or False,
+            })
+            try:
+                partner_dummy.check_vat()
+            except exceptions.ValidationError as err:
+                error = 'vat_not_valid'
+                message = err.name
+        return error, message
+
+    def form_before_create_or_update(self, values, extra_values):
+        user = self.env.user
+        # handle email update
+        if 'email' in values and user.email != values.get('email'):
+            self._handle_email_update(user, values)
+
+    def _handle_email_update(self, user, values):
+        """Validate email update and handle login update.
+
+        :param user: res.users record
+        :param values: form request values
+
+        If email is validated:
+
+            * check if already exists
+            * if yes, provide an error message
+            * if not, updates user's login
+            * provide an info message
+            * logout current session to force re-login with new email
+        """
+        email = values['email']
+        # TODO: this could be useless as we validate w/ `form_validate_email`
+        valid = validate_email(email)
+        if email and valid and not user._is_admin():
+            exists = user.sudo().search_count(
+                ['|', ('login', '=', email), ('email', '=', email)]
+            )
```

Comment:
Should inactive users be searched as well? I don't really know what'd happen if an inactive user exists with the same email and you don't detect it (nor how we should handle that)

Reply:
there's an SQL constrain so it should fail badly. Probably is better if we catch that too.